### PR TITLE
fix: client app retry logic

### PR DESCRIPTION
**Pull request by braginini**

### CODE REVIEW

File: management/client/client.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (c *Client) Sync(msgHandler func(msg *proto.SyncResponse) error) error {
 			log.Errorf("failed to open Management Service stream: %s", err)
 			return err
 		}
-
+		backOff.Reset()
```
Comment (mlsmaycon) [inline management/client/client.go]: This won't have the desired effect if we block the connection to receive events for more than the max elapsed time

Comment (mlsmaycon) [reply management/client/client.go]: perhaps we should move/add it to line 106

File: signal/client/client.go [hunk 2]
```diff
@@ -101,14 +101,19 @@ func (c *Client) Receive(msgHandler func(msg *proto.Message) error) {
...
-			err := c.connect(c.key.PublicKey().String(), msgHandler)
+			stream, err := c.connect(c.key.PublicKey().String())
 			if err != nil {
 				log.Warnf("disconnected from the Signal Exchange due to an error: %v", err)
 				c.connWg.Add(1)
 				return err
 			}
 
 			backOff.Reset()
+			err = c.receive(stream, msgHandler)
```
Comment (mlsmaycon) [inline signal/client/client.go:111]: Same comment as for management client, if we reset things before blocking we risk facing the same issue. We can add backOff.Reset() in the error treatment as well

File: management/client/client.go (hunk 1 above)
Comment (braginini) [reply management/client/client.go]: done

File: signal/client/client.go (hunk 2 above)
Comment (braginini) [reply signal/client/client.go:111]: done

**Review by mlsmaycon — APPROVED** [review]

**Pull request merged**